google/cloud/opentelemetry/internal/time_series_test.cc

```diff
@@ -697,6 +711,82 @@ TEST(ToRequests, Batches) {
   }
 }
 
+TEST(WithExtraLabels, CopyLabels) {
+  opentelemetry::sdk::metrics::PointDataAttributes pda;
+  pda.point_data = opentelemetry::sdk::metrics::SumPointData{};
+
+  opentelemetry::sdk::metrics::MetricData md;
+  md.point_data_attr_.push_back(std::move(pda));
+  md.instrument_descriptor.name_ = "metric-name";
+  md.instrument_descriptor.unit_ = "unit";
+  md.instrument_descriptor.type_ = {};
+  md.instrument_descriptor.value_type_ = {};
+
+  opentelemetry::sdk::metrics::ScopeMetrics sm;
+  sm.metric_data_.push_back(md);
+  sm.metric_data_.push_back(std::move(md));
+
+  opentelemetry::sdk::metrics::ResourceMetrics rm;
+  auto resource = ResourceWithAllExtraLabels();
+  rm.resource_ = &resource;
+  rm.scope_metric_data_.push_back(std::move(sm));
+
+  auto tss = ToTimeSeries(rm, PrefixWithWorkload);
+  for (auto const& kv : kExtraLabelsLookup) {
+    auto const& oks = kv.second.otel_keys;
+    if (oks.empty()) {
+      continue;
+    }
+    for (auto const& ts : tss) {
+      auto const& labels = ts.metric().labels();
+      ASSERT_TRUE(labels.find(kv.first) != labels.end());
+      EXPECT_EQ(labels.at(kv.first), kv.first);
```

Comment:
We probably want to use GMock's container matchers.

https://google.github.io/googletest/reference/matchers.html